index.d.ts

```diff
@@ -0,0 +1,61 @@
+import { BrowserWindow } from "electron";
+
+export interface Options {
+	/**
+	 * Default: [Only in development](https://github.com/sindresorhus/electron-is-dev)
+	 */
+	enabled?: boolean;
+
+	/**
+	 * Show DevTools on each created `BrowserWindow`.
+	 *
+	 * @default true
+	 */
+	showDevTools?: boolean;
+
+	/**
+	 * The dock state to open DevTools in.
+	 *
+	 * @default 'undocked'
+	 */
+	devToolsMode?: 'undocked' | 'right' | 'bottom' | 'previous' | 'detach';
+}
+
+/**
+ * Install keyboard shortcuts and optionally activate DevTools on each created `BrowserWindow`.
+ *
+ * @example
+ *
+ * import {app, BrowserWindow} from 'electron';
+ * import electronDebug from 'electron-debug';
+ *
+ * electronDebug();
+ *
+ * let win;
+ * (async () => {
+ * 	await app.whenReady();
+ * 	win = new BrowserWindow();
+ * });
+ */
+export default function electronDebug(options?: Readonly<Options>): void;
```

Comment:
Hmm, I think we should use `readonly` on the specific properties whenever possible, as `Readonly` will show up in the code completion and create noise: <img width="336" alt="screen shot 2019-01-28 at 19 56 25" src="https://user-images.githubusercontent.com/170270/51837727-e2b17e80-2336-11e9-8f32-bb90903026cd.png">

Maybe that's a non-issue. I dunno.

Thoughts?

---

*Really want [the `readonly` type operators](https://github.com/Microsoft/TypeScript/issues/10725)...

Reply:
I'm not sure either…
Is that information useful in any way to the end user of the API? If it's not, then sure, `readonly` all the props. 😉

Reply:
It’s not. It’s only useful for us to ensure we don’t modify the object.

Reply:
Let’s go with per-property annotation.